parser.y

```diff
@@ -5275,16 +5275,19 @@ NotKeywordToken:
  *  TODO: support PARTITION
  **********************************************************************************/
 InsertIntoStmt:
-	"INSERT" PriorityOpt IgnoreOptional IntoOpt TableName InsertValues OnDuplicateKeyUpdate
+	"INSERT" TableOptimizerHints PriorityOpt IgnoreOptional IntoOpt TableName InsertValues OnDuplicateKeyUpdate
 	{
-		x := $6.(*ast.InsertStmt)
-		x.Priority = $2.(mysql.PriorityEnum)
-		x.IgnoreErr = $3.(bool)
+		x := $7.(*ast.InsertStmt)
+		x.Priority = $3.(mysql.PriorityEnum)
+		x.IgnoreErr = $4.(bool)
 		// Wraps many layers here so that it can be processed the same way as select statement.
-		ts := &ast.TableSource{Source: $5.(*ast.TableName)}
+		ts := &ast.TableSource{Source: $6.(*ast.TableName)}
 		x.Table = &ast.TableRefsClause{TableRefs: &ast.Join{Left: ts}}
-		if $7 != nil {
-			x.OnDuplicate = $7.([]*ast.Assignment)
+		if $8 != nil {
+			x.OnDuplicate = $8.([]*ast.Assignment)
+		}
+		if $2 != nil {
+			x.TableHints = $2.([]*ast.TableOptimizerHint)
```

Comment:
what about this?

```sql
insert /*+ qb_name(qb1) */ into a select /*+ qb_name(qb2) */ 1;
```

MySQL 8.0 currently treats this as the same as `/*+ qb_name(qb2) qb_name(qb1) */`.

Reply:
good catch. What about ignoring `TableOptimizerHint` when `TableHints` is defined in `InsertValues`?

Reply:
We would store the hints in each own stmtNode and solve the hints priority in TiDB.